src/randoop/operation/ConstructorCall.java

```diff
@@ -149,32 +146,39 @@ public void appendCode(Variable varName, List<Variable> inputVars, StringBuilder
       // or string literal, like "int x = 3" are not added to a sequence;
       // instead, the value (e.g. "3") is inserted directly added as
       // arguments to method calls.
-      Operation statementCreatingVar = inputVars.get(i).getDeclaringStatement(); 
-      if (!GenInputsAbstract.long_format
-          && ExecutableSequence.canUseShortFormat(statementCreatingVar)) {
-        b.append(PrimitiveTypes.toCodeString(((NonreceiverTerm) statementCreatingVar).getValue()));
+      Statement statementCreatingVar = inputVars.get(i).getDeclaringStatement(); 
+      String shortForm = statementCreatingVar.getShortForm();
+      if (!GenInputsAbstract.long_format && shortForm != null) {
+        b.append(shortForm);
```

Comment:
Here the test seems right (but it's a touch inefficient to generate the short form whether or not it will be needed, so that is a negative).